extensions/Xeltalliv/simple3D.js

```diff
@@ -0,0 +1,3808 @@
+			text: "all meshes",
+			disableMonitor: true,
+			def: function() {
+				return Array.from(meshes.keys()).join(",");
+			}
+		},
+		{
+			opcode: "createMesh",
+			blockType: BlockType.COMMAND,
+			text: "create mesh [NAME]",
+			arguments: {
+				NAME: {
+					type: ArgumentType.STRING,
+					defaultValue: "my mesh"
+				},
+			},
+			def: function({NAME}) {
+				NAME = Cast.toString(NAME);
+				if (NAME.length == 0) return;
+				meshes.get(NAME)?.destroy();
+				meshes.set(NAME, new Mesh(NAME));
+			}
+		},
+		{
+			opcode: "deleteMesh",
+			blockType: BlockType.COMMAND,
+			text: "delete mesh [NAME]",
+			arguments: {
+				NAME: {
+					type: ArgumentType.STRING,
+					defaultValue: "my mesh"
+				},
+			},
+			def: function({NAME}) {
+				NAME = Cast.toString(NAME);
+				meshes.get(NAME)?.destroy();
+				meshes.delete(NAME);
+			}
+		},
+		{
+			opcode: "inheritMeshes",
+			blockType: BlockType.COMMAND,
+			text: "make [NAME] inherit from meshes [NAMES]",
+			arguments: {
+				NAME: {
+					type: ArgumentType.STRING,
+					defaultValue: "my mesh 3"
+				},
+				NAMES: {
+					type: ArgumentType.STRING,
+					defaultValue: "my mesh 1,my mesh 2"
+				},
+			},
+			def: function({NAME, NAMES}) {
+				const mesh = meshes.get(Cast.toString(NAME));
+				if (!mesh) return;
+				const parentMeshes = Cast.toString(NAMES).split(",").map(s => meshes.get(s.trim())).filter(m => m);
+				for(let otherMesh of parentMeshes) {
+					if (otherMesh.dependsOn(mesh)) return;
+				}
+				mesh.dependencies = new Set(parentMeshes);
+				for(let otherMesh of parentMeshes) {
+					otherMesh.dependants.add(mesh);
+				}
+				mesh.update();
+			}
+		},
+		{
+			opcode: "meshInfo",
+			blockType: BlockType.REPORTER,
+			text: "mesh [NAME] [PROP]",
+			allowDropAnywhere: true,
+			arguments: {
+				PROP: {
+					type: ArgumentType.STRING,
+					menu: "meshProperties",
+					defaultValue: "inherits from"
+				},
+				NAME: {
+					type: ArgumentType.STRING,
+					defaultValue: "my mesh"
+				},
+			},
+			def: function({NAME, PROP}) {
+				const mesh = meshes.get(Cast.toString(NAME));
+				if (PROP == "exists") return !!mesh;
+				if (!mesh || !hasOwnProperty.call(MeshPropFns, PROP)) return "";
+				return MeshPropFns[PROP](mesh) ?? "";
+			}
+		},
+		"---",
+		{
+			opcode: "setMeshIndices",
+			blockType: BlockType.COMMAND,
+			text: "set [NAME] vertex indices [INDICES]",
+			arguments: {
+				NAME: {
+					type: ArgumentType.STRING,
+					defaultValue: "my mesh"
+				},
+				INDICES: {
+					type: ArgumentType.STRING,
+					menu: "lists"
+				},
+			},
+			def: function({NAME, INDICES}, {target}) {
+				const mesh = meshes.get(Cast.toString(NAME));
+				const value = compactIndices(target, INDICES);
+				if (!mesh || !value) return;
+				uploadBuffer(mesh, "indices", value, 1, -1, gl.ELEMENT_ARRAY_BUFFER);
+			}
+		},
+		{
+			opcode: "setMeshPositionsXY",
+			blockType: BlockType.COMMAND,
+			text: "set [NAME] positions XY [X] [Y]",
```

Comment:
The reason it was combined together is because otherwise it's hard to tell apart similar blocks with different amount of arguments, or blocks in general. Currently it's very obvious at a glance.
![image](https://github.com/TurboWarp/extensions/assets/90340988/65223519-92b1-4cca-bb54-369a78dfeca4)

Reply:
I didn't think of that, that makes sense